### PR TITLE
API Available registration methods are now rendered with their thumbnails and information

### LINKED ISSUE #34: Option to select authentication method
As a CMS user registering a MFA method, I want to be able to chose which method to set up from available options, so that I can select my preferred method.

Designs: https://projects.invisionapp.com/share/3PNSKZQYBJZ#/screens/321189815

ACs
- There is a selected state for verification methods
- The template has an area to display information on the authentication methods (including 'Find out more' link) which will be surfaced from the authentication method's own components
- The 'Next' button is disabled until a selection is made
- Styling is created, as per designs, to disable authentication methods that may not be available. This would be attainted from the method module. An example being U2F is only supported by default in Chrome v71+ so it would otherwise be disabled in another browser
- The user has the ability to go back to the previous screen
- Area has the ability to stack up to three 'method' boxes wide
- Create default styling within the app, using generic class names that could be overwritten by a site developer

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #34

We've decided that since more than 3 verification methods is unlikely in a project, we won't bother trying to constrain the designs too much to fit into the same column as the first row's first method. This is designed to handle two or three authenticators nicely. Project code can adjust the CSS for this easily to make four look good in two rows or two (for example) if they want to.

One method:

image

One method selected:

image

Two methods:

image

Three methods:

image

Hover/focus state:

image

Unavailable/unsupported state (with mock data):

image

Four methods (unlikely):

image

Five methods (unlikely):

image

Six methods (unlikely):

image

### CODE REVIEW

File: client/src/components/Register/MethodTile.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    width: $tile-thumbnail-size;
+  }
+
+  &__title {
+    font-size: $font-size-h1;
+  }
+
+  &__unavailable {
+    &-mask {
+      background: rgba(#ffffff, .75);
```
Comment (robbieaverill) [inline client/src/components/Register/MethodTile.scss]: Should probably be a variable

File: client/src/components/Register/MethodTile.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+  }
+
+  /**
+   * If the method is enabled, trigger the click handler prop
+   *
+   * @param event
```
Comment (robbieaverill) [inline client/src/components/Register/MethodTile.js]: Missing type

File: client/src/components/Register/MethodTile.scss [hunk 3]
```diff
@@ -0,0 +1,64 @@
+.mfa-method-tile {
+  flex: 25%;
```
Comment (robbieaverill) [inline client/src/components/Register/MethodTile.scss]: Why isn't this 33%

File: client/src/components/Register/SelectMethod.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  handleGoToNext() {
+    const { highlightedMethod } = this.state;
+
+    this.props.onSelectMethod(highlightedMethod);
+  }
+
+  /**
+   * Handle clicking on a method
+   *
+   * @param {object} method
```
Comment (robbieaverill) [inline client/src/components/Register/SelectMethod.js]:
```suggestion
   * @param {availableMethodType} method
```

File: client/src/components/Register/MethodTile.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          role="button"
+        >
+          {method.thumbnail && (
+            <div className="mfa-method-tile__thumbnail-container">
+              <img src={method.thumbnail} className="mfa-method-tile__thumbnail" alt={method.name} />
+            </div>
+          )}
+          <h3 className="mfa-method-tile__title">{method.name}</h3>
+          <p className="mfa-method-tile__description">
+            {method.description}. {this.renderSupportLink(method)}
```
Comment (NightJar) [inline client/src/components/Register/MethodTile.js]: is the description a required property of `method`? The `.` seems a bit out of place if it's not. Or could cause `My description..` if it is...

File: client/src/components/Register/MethodTile.scss (hunk 3 above)
Comment (robbieaverill) [reply client/src/components/Register/MethodTile.scss]: Because 33% including margins breaks early. I'll update to `calc(33% - 1rem);`

**Comment by robbieaverill** [comment]
Also missing media queries for smaller viewports

File: client/src/components/Register/MethodTile.scss (hunk 3 above)
Comment (ScopeyNZ) [reply client/src/components/Register/MethodTile.scss]: Try `calc(33% - (#{$tile-spacing} / 2))`

File: client/src/components/Register/MethodTile.scss [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  margin-right: $tile-spacing;
```
Comment (ScopeyNZ) [inline client/src/components/Register/MethodTile.scss]:
```suggestion
```

By using `calc(33% - (#{$tile-spacing} / 2))` above and putting justify-content in the container you get this spacing done by flex (and no extra margin at the end of the row)

File: client/src/components/Register/MethodTile.scss (hunk 3 above)
Comment (ScopeyNZ) [inline client/src/components/Register/MethodTile.scss]:
```suggestion
  flex-basis: calc(33% - (#{$tile-spacing} / 2));
```

File: client/src/components/Register/SelectMethod.scss [hunk 7]
```diff
@@ -0,0 +1,15 @@
+.mfa-method-tile {
+  &__container {
+    display: flex;
+    flex-flow: row wrap;
+    list-style-type: none;
+    padding: 0;
+    width: 100%;
```
Comment (ScopeyNZ) [inline client/src/components/Register/SelectMethod.scss]:
```suggestion
    width: 100%;
    justify-content: space-between;
```

**Comment by ScopeyNZ** [comment]
I have a semantics suggestion for the 4 authenticators scenario. You can apply a class in React:

__SelectMethod.js:78__
```
    // Specify 2 columns if theres a number of methods that's a multiple of 2 but not also 3.
    const containerClass = classNames('mfa-method-tile__container', {
      'two-columns': (methods.length % 2 === 0) && (methods.length % 3 !== 0)
    });
```

And then adjust the styling to use this:

__MethodTile.scss:7__
```
  .two-columns & {
    flex-basis: calc(50% - (#{$tile-spacing} / 2));
  }
```

**Pull request merged**